src/network/networkctl.c

```diff
@@ -2552,28 +2552,30 @@ static int link_lldp_status(int argc, char *argv[], void *userdata) {
                         (void) sd_lldp_neighbor_get_system_name(n, &system_name);
                         (void) sd_lldp_neighbor_get_port_description(n, &port_description);
 
-                        if (chassis_id) {
-                                cid = ellipsize(chassis_id, 17, 100);
-                                if (cid)
-                                        chassis_id = cid;
-                        }
+                        if (!arg_full) {
+                                if (chassis_id) {
+                                        cid = ellipsize(chassis_id, 17, 100);
```

Comment:
In the above, we have
```
        if (arg_full)
                table_set_width(table, 0);
```
and table_add_many() ellipsizes strings gracefully.

Hence, please unconditionally put the given strings, e.g. chassis_id, to the table.

Reply:
Oh, so you mean all this ellipsizing code from line 2555 to 2578 is superfluous since `table_add_many()` ellipsizes strings anyway? I just tried it by commenting those lines out but nothing is ellipsized in my table.

Reply:
See,
https://github.com/systemd/systemd/blob/c87c30780624df257ed96909a2286b2b933f8c44/src/shared/format-table.c#L136-L137
And, by default, the width is SIZE_MAX. Hence, each strings are truncated if you run the command on a terminal with short width.

Reply:
Please provide the diff you tested.

Reply:
This:

```
diff --git a/src/network/networkctl.c b/src/network/networkctl.c
index d656027ec4..d2a72dec27 100644
--- a/src/network/networkctl.c
+++ b/src/network/networkctl.c
@@ -2551,7 +2551,7 @@ static int link_lldp_status(int argc, char *argv[], void *userdata) {
                         (void) sd_lldp_neighbor_get_port_id_as_string(n, &port_id);
                         (void) sd_lldp_neighbor_get_system_name(n, &system_name);
                         (void) sd_lldp_neighbor_get_port_description(n, &port_description);
-
+/*
                         if (chassis_id) {
                                 cid = ellipsize(chassis_id, 17, 100);
                                 if (cid)
@@ -2575,7 +2575,7 @@ static int link_lldp_status(int argc, char *argv[], void *userdata) {
                                 if (pdesc)
                                         port_description = pdesc;
                         }
-
+*/
                         if (sd_lldp_neighbor_get_enabled_capabilities(n, &cc) >= 0) {
                                 capabilities = lldp_capabilities_to_string(cc);
                                 all |= cc;

```

None of the columns is truncated when executing `ninja -C build/ && ./build/networkctl lldp`, the table overflows my terminal. Terminal widths:

```
[nix-shell:~/systemd]$ tput cols
60
```

Reply:
Maybe, `table_set_minimum_width()` in the above is too large for your terminal width?

Reply:
Even with a width of `145` the table overflows as no string in any column is truncated. I have one very long system name. Please try it out yourself, the column formatting definitely works not as expected.